types/three/examples/jsm/nodes/core/NodeBuilder.d.ts

```diff
@@ -0,0 +1,125 @@
+import {
+    BufferGeometry,
+    Material,
+    Object3D,
+    Renderer,
+    Scene,
+    Texture,
+    TextureEncoding,
+    WebGLRenderTarget,
+} from '../../../../src/Three';
+import FogNode from '../fog/FogNode';
+import LightsNode from '../lighting/LightsNode';
+import { NodeShaderStageOption, NodeTypeOption } from './constants';
+import Node from './Node';
+import NodeAttribute from './NodeAttribute';
+import NodeParser from './NodeParser';
+import NodeUniform from './NodeUniform';
+import NodeVar from './NodeVar';
+import NodeVary from './NodeVary';
+
+export type BuildStageOption = 'construct' | 'analyze' | 'generate';
+
+export interface FlowData {
+    code: string;
+}
+
+export default class NodeBuilder {
+    object: Object3D;
+    material: Material;
```

Comment:
I don't know enough about `node` but from experience with things like `mesh` i'm wondering if a generic would be beneficial here to accurately depict the `Material` (same with BufferGeometry, Renderer & NodeParser)

Reply:
You are probably right, but I would opt for deleting the property from typings or just using a union type, `NodeBuilder` will not be used by majority anyways. And making it generic would genericize everything.